src/main/java/com/sleekbyte/tailor/output/Printer.java

```diff
@@ -146,7 +163,13 @@ public int hashCode() {
     }
 
     private void print(Rules rule, Severity severity, String msg, Location location) {
-        ViolationMessage violationMessage = new ViolationMessage(rule, location.line, location.column, severity, msg);
+        ViolationMessage violationMessage;
+        if (rule != null) {
```

Comment:
Let's just use an overloaded function `print(Severity, String, Location)` and not have this check? Or alternatively, should we just create a rule for the on-off mismatch? If we are storing the warning using `ViolationMessage`, I think we should.